lib/facility_builder.rb

```diff
@@ -0,0 +1,52 @@
+class FacilityBuilder
+  def build_facility_data(dmv_data)
+    facilities = []
+    dmv_data.each do |dmv_entry|
+      
+      
+      entry_name = nil
```

Comment:
Great job figuring out how to handle data from multiple sources!  You could also move some of this logic to helper methods - you could imagine that all 50 states might have different formats, and that would make this method really difficult to read through.  That refactor might look like:

```ruby
entry_name = set_name_attribute(dmv_entry)

...

def set_name_attribute(dmv_entry)
  dmv_entry[:dmv_office] || dmv_entry[:office_name] || dmv_entry[:name]
end